mtags/src/main/scala-0.24/scala/scala/meta/internal/pc/CompilerInterfaces.scala

```diff
@@ -0,0 +1,94 @@
+package scala.meta.internal.pc
+
+import java.net.URI
+import java.nio.file.Paths
+import java.io.BufferedWriter
+import java.io.OutputStreamWriter
+import java.{util => ju}
+import scala.collection.JavaConverters._
+import scala.io.Codec
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.reporting.StoreReporter
+import dotty.tools.dotc.parsing.Parsers.Parser
+import dotty.tools.dotc.reporting.Message
+import dotty.tools.dotc.reporting.StoreReporter
+import dotty.tools.dotc.reporting.Diagnostic.Error
+import dotty.tools.dotc.interfaces.Diagnostic
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.util.SourcePosition
+import dotty.tools.io.VirtualFile
+import org.eclipse.{lsp4j => l}
+
+// note(@tgodzik) the plan is to be able to move the methods here back to Dotty compiler
+// so that we can provide easier compatibility with multiple Scala 3 versions
+object CompilerInterfaces {
```

Comment:
So this is my first attempt to move some of the methods we use to a separate class. In time we might be able to move those methods into Dotty itself and allow for better compatibility with different Scala versions.

Plus this part was incompatible between 0.23 and 0.24